app/pubhub/v1/user/loans/[identifier]/route.ts

```diff
@@ -0,0 +1,34 @@
+import { NextRequest, NextResponse } from "next/server"
+
+import { withAuth } from "@/app/pubhub/(lib)/helper"
+import { TUserInfo } from "@/app/pubhub/(lib)/types"
+import { getLastValueFromUrl, transformTimeToUtcString } from "@/app/pubhub/helper"
+
+import { createLoanRequest } from "./(lib)/requests"
+import { createLoanSchema } from "./(lib)/schemas"
+
+async function postLibraryLoan(request: NextRequest, context: { userInfo: TUserInfo }) {
+  const createLoan = createLoanSchema.transform(loanData => {
+    const response = loanData.response
+    return {
+      expirationDateUtc: transformTimeToUtcString(response.data.expirationdateutc),
+      orderNumber: response.data.internalordernumber,
+      orderId: response.data.retailerordernumber,
+      code: response.status.code,
+      message: `${response.status.message} (#${response.status.code})`,
+    }
+  })
+  try {
+    const responseData = await createLoanRequest(context.userInfo, getLastValueFromUrl(request.url))
```

Comment:
You should get the identifier via the an argument in  the `postLibraryLoan` function. You can find documentation about it in the Nextjx documentation somewhere around dynamic routes.

Reply:
Problem there ([see example](https://nextjs.org/docs/pages/building-your-application/routing/dynamic-routes#example)) is that the useRouter is a hook and hooks can't be called inside functions, only components. Do you still think this is insufficient in that case?

Reply:
I did that now :))